### DISCUSSION TITLE
Spreading the particles evenly in all directions.

### DISCUSSION THREAD

**Discussion by Dhadzo** [matteobruni/tsparticles#1362] [General]
I'm using the react-tsparticles library to display a circle of particles. The problem is that there is more particles on one side of the circle compared to the other. What property can I use to distribute them equally in the circle.

**Comment by matteobruni** [comment]
Hi @Dhadzo,

Can you share the configuration you are using? There are some solutions to this problem but I’d like to choose the one that fits the most

**Comment by Dhadzo** [comment]
Hi @matteobruni , thanks for your response.  Here's the configuration 

```
   <Particles
	id="tsparticles"
	width={1250}
	height={1000}
	options={{
		background: {
			color: {
				value: '#ffffff',
			},
			opacity: '0.4',
		},
		detectRetina: false,
		fpsLimit: 60,
		interactivity: {
			detectsOn: 'canvas',
			events: {
				onHover: {
					enable: true,
					mode: 'bubble',
				},
				resize: true,
			},
			modes: {
				bubble: {
					distance: 10,
					duration: 2,
					opacity: 8,
					size: 10,
					speed: 3,
				},
			},
		},
		position: {
			x: 500,
			y: 300,
		},
		particles: {
			color: {
				value: '#FFB782',
			},
			collisions: {
				bounce: {
					horizontal: {
						random: {
							enable: true,
							minimumValue: 0.1,
						},
						value: 1,
					},
					vertical: {
						random: {
							enable: true,
							minimumValue: 0.1,
						},
						value: 1,
					},
				},
				enable: false,
				mode: 'bounce',
			},
			lineLinked: {
				blink: false,
				color: 'random',
				consent: false,
				distance: 170,
				enable: true,
				collissions: false,
				width: 1,
			},
			reduceDuplicates: true,
			move: {
				attract: {
					enable: false,
					rotate: {
						x: 1200,
						y: 1200,
					},
				},
				bounce: false,
				direction: 'clockwise',
				enable: true,
				outMode: 'bounce',
				random: false,
				speed: 1,
				straight: false,
			},
			angle: {
				offset: 30,
				value: 60,
			},
			number: {
				density: {
					enable: false,
					area: 800,
					factor: 2000,
				},
				limit: 200,
				value: 50,
			},
			opacity: {
				random: false,
				value: 1,
			},
			shape: {
				options: {
					character: {
						fill: false,
						font: 'Verdana',
						weight: '400',
					},
					char: {
						fill: false,
						font: 'Verdana',
						weight: '400',
					},
					polygon: {
						sides: 5,
					},
					star: {
						sides: 5,
					},
				},
				type: 'circle',
			},
			size: {
				animation: {
					enable: false,
					minimumValue: 0.1,
					speed: 40,
					sync: false,
				},
				random: false,
				value: 20,
			},
		},
		backgrounMask: {
			composite: 'destination-out',
			cover: {
				color: {
					value: '432423',
				},
				opacity: '1',
			},
			enable: true,
		},
		polygon: {
			draw: {
				enable: false,
				lineColor: '#FFB782',
				lineWidth: 0.3,
			},
			move: {
				radius: 30,
			},
			inlineArrangement: 'equidistant',
			scale: 2,
			url: 'path-to-svg',
		},
	}}
/>;
```

**Reply by matteobruni** [reply]
I don't understand where the circle is. I thought you had a polygon mask with particles inside.

Do you have some CSS to show only a circular region of the particles div?

**Reply by Dhadzo** [reply]
My apologies its not exactly  a circle. I'm trying to create something like this: https://raw.githubusercontent.com/Dhadzo/portfolio/c8be367ca2906b9fae0f66f1d3326038c2c80f54/header.svg

Here's is what I've done so far: 
https://codesandbox.io/s/laughing-chatterjee-2c9sv?file=/src/App.js

**Reply by matteobruni** [reply]
I've updated the sample: https://codesandbox.io/s/empty-dew-8qwcy

The polygon mask was not used since you didn't specify a `type` (by default is `none` and it ignores the feature). I used the type `inside` so particles will be inside the specified SVG.

If you want to place particles manually you can checkout these samples:

https://codepen.io/matteobruni/pen/GRpYdGv -> generated particles with javascript and added dynamically

https://codepen.io/matteobruni/pen/YzpOZpq -> manual particles added by config, easier but it's more static.

**Reply by Dhadzo** [reply]
Thank you @matteobruni, the solution of adding particles manually is exactly what I was looking for.
